controllers/argocd/metrics.go

```diff
@@ -0,0 +1,147 @@
+package argocd
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// workloadState is a named int type that is used to characterize the state (unknown/pending/running/available) a workload is in
+type workloadState int
+
+// numerical representation of workloadStates to encode this information into prometheus
+const (
+	WorkloadUnknownState   workloadState = 0
+	WorkloadFailedState    workloadState = 1
+	WorkloadPendingState   workloadState = 2
+	WorkloadRunningState   workloadState = 3
+	WorkloadAvailableState workloadState = 4
+)
+
+const (
```

Comment:
This can be added into the above constants block. We can distinuish the workload related constants from this one using some comments.

Reply:
done, thanks